packages/start/config/index.d.ts

```diff
@@ -9,7 +9,14 @@ type SolidStartInlineConfig = Omit<InlineConfig, "router"> & {
         appRoot?: string,
         middleware?: string,
         islands?: boolean
+    },
+    plugins?: Plugins | {
+        client?: Plugins,
+        server?: Plugins,
+        serverFunctions?: Plugins
     }
 }
 
+type Plugins = InlineConfig["plugins"] | (() => Promise<InlineConfig["plugins"]> | InlineConfig["plugins"])
```

Comment:
you have `InlineConfig["plugins"]` twice?

Reply:
That's interpreted as `T | (() => (Promise<T> | T))`